### ISSUE TITLE
Message history request after Twitch RECONNECT message is unecessarily large

### ISSUE THREAD

**Issue by RAnders00** [Chatterino/chatterino2#4989]
### Checklist

- [X] I'm reporting a problem with Chatterino
- [X] I've verified that I'm running **the most** recent nightly build or stable release
- [X] I've looked for my problem on the [wiki](https://wiki.chatterino.com/Help/)
- [X] I've searched the [issues and pull requests](https://github.com/Chatterino/chatterino2/issues?q=) for similar looking reports

### Describe your issue

I run the [recent-messages](https://recent-messages.robotty.de/) service that chatterino integrates with by default.
I have noticed an issue that arises when Twitch sends `RECONNECT` messages to all its users: My service gets bombarded with uncessarily taxing requests. The default `?limit=800` parameter is used, even though the disconnect timeframe is arguably very small.

I'd like to request that in this special case, (and probably in case of all reconnects), the `?limit=` parameter be chosen dynamically based on the time duration that the client was disconnected. Maybe something like "10 for every second spent disconnected" is reasonable. Reducing the `?limit` reduces the amount of CPU work my service has to do, by a significant amount.

I don't want to scale my service for the event that the whole world of chatterino requests all their channels at once, but I also want to provide a good service, i.e. requests should generally succeed, henceforth my request.

For illustration, some metrics:

![image](https://github.com/Chatterino/chatterino2/assets/1629196/fe518be8-985e-47c2-b4d9-bb72927ee9f3)
![image](https://github.com/Chatterino/chatterino2/assets/1629196/12d1affd-cf0f-4241-9349-d7e26cc06179)
![image](https://github.com/Chatterino/chatterino2/assets/1629196/699beb09-cf84-4e37-ac89-337c301e609d)
![image](https://github.com/Chatterino/chatterino2/assets/1629196/e0669412-4cae-4cad-885c-382ccd66b358)

```none
2023/11/29 18:16:00 [alert] 401720#401720: *36739 socket() failed (24: Too many open files) while connecting to upstream, client: [redacted], server: recent-messages.robotty.de, request: "GET /api/v2/recent-messages/[redacted]?limit=800 HTTP/2.0", upstream: "http://unix:/var/run/recent-messages2/server.sock:/api/v2/recent-messages/[redacted]?limit=800", host: "recent-messages.robotty.de"
```

Over 100000 requests didnt even reach my application and thus are not even represented in the request rate graph:

```
❯ grep "Too many open files" /var/log/nginx/error.log | wc -l
117125
```

### OS and Chatterino Version

Most requests are seen from 2.4.4, 2.4.5, 2.4.6 and 7.4.6

**Comment by RAnders00** [comment]
Note, I will look into implementing a `?since=` parameter, which would be most appropriate here. I'm tracking this as https://github.com/robotty/recent-messages2/issues/286

**Comment by pajlada** [comment]
Fixed by #5001  - some more improvements are being discussed in #5009
